src/Job/DeveloperSync.php

```diff
@@ -59,128 +58,128 @@ class DeveloperSync extends EdgeJob {
    * DeveloperSync constructor.
    *
    * @param null|string $filter
-   *   An optional regexp filter for the edge developer emails.
+   *   An optional regexp filter for the Apigee Edge developer emails.
    */
   public function __construct(?string $filter) {
     parent::__construct();
     $this->filter = $filter;
   }
 
   /**
-   * Returns the database connection service.
+   * Loads all Drupal users indexed my their emails.
    *
-   * @return \Drupal\Core\Database\Connection
-   *   The database connection service.
+   * @return \Drupal\user\UserInterface[]
+   *   Format: strtolower(mail) => User
    */
-  protected function getConnection() : Connection {
-    return \Drupal::service('database');
-  }
-
-  /**
-   * Loads all users' emails.
-   *
-   * @return array
-   *   Format: strtolower(mail) => mail
-   */
-  protected function loadUserEmails() : array {
-    $mails = $this->getConnection()->query("
-      SELECT u.mail
-      FROM {users_field_data} u
-    ")->fetchCol();
-
-    $accounts = [];
-    foreach ($mails as $mail) {
-      if (isset($mail)) {
-        $accounts[strtolower($mail)] = $mail;
+  protected function loadUsers(): array {
+    $users = [];
+    /** @var \Drupal\user\UserInterface $user */
+    foreach (User::loadMultiple() as $user) {
+      $email = $user->getEmail();
+      if (isset($email)) {
+        $users[strtolower($email)] = $user;
```

Comment:
Let's leave a reminder here that we did `strtolower($email)` because of this: "https://www.drupal.org/project/drupal/issues/2490294".